File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    stack = []
    
    for i, word in enumerate(words):
        if word in stack or (stack and stack[-1][-1] != word[0]):
            
            return [(i % n) + 1, (i // n) + 1]
        else:
            stack.append(word)
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];
        HashSet<String> map = new HashSet<>();
        map.add(words[0]);
        String before = words[0];
        
        for (int i=1; i < words.length; i++) {
            int bl = map.size();
            String word = words[i];
            map.add(word);
            int al = map.size();
            
            if (before.charAt(before.length()-1) == word.charAt(0) && bl != al) {
                before = word;
            } else {
                answer[0] = (i%n)+1;
                answer[1] = (int)(Math.ceil((double)(i+1)/n));
                return answer;
            }
        }
        return answer;
    }
}